libs/utils/env.py

```diff
@@ -439,6 +439,51 @@ def _init_target(self, force = False):
             self.__modules = ['bl', 'cpufreq']
             platform = Platform(model='pixel')
 
+        # Initialize gem5 platform
+        elif self.conf['board'].upper() == 'GEM5':
+            self.nrg_model = None
+            self.__modules=['cpufreq']
+
+            system_info = self.conf['gem5_settings']['system']
+            simulator_info = self.conf['gem5_settings']['simulator']
+            simulator_args = ''
+            platform_args = ''
+
+            # Get gem5 binary arguments
+            simulator_args = ' '.join(simulator_info.get('args', []) +
+                                      ['--listener-mode=on'])
+
+            # Get platform arguments
+            platform_args = ' '.join(system_info['platform'].get('args', []) +
+                                     ['--kernel {}'.format(system_info['kernel']),
+                                      '--dtb {}'.format(system_info['dtb']),
+                                      '--disk-image {}'.format(system_info['disk'])])
+
+            # Gather all arguments
+            args = ' '.join([simulator_args,
+                             system_info['platform']['description'],
+                             platform_args])
+
+            diod_path = which('diod')
+            if diod_path is None:
+                raise RuntimeError('Failed to find diod on your host machine, '
+                                   'check your installation or your PATH variable')
+
+            # Setup virtio
+            virtio_args = ' '.join(['--which-diod=' + diod_path,
+            # Brackets are there to let the output dir be created automatically
+                                    '--workload-automation-vio={}'])
+
+            # Merge all arguments
+            platform = devlib.platform.gem5.Gem5SimulationPlatform(
+                name='gem5',
+                gem5_bin=simulator_info['gem5_bin'],
```

Comment:
The GEM5 binaries used are gem5.fast, gem5.opt, etc., depending on how GEM5 was built and they won't be present in path so it's not useful to have this. It's also not useful to have a default as it depends on how the user built GEM5 and it's up to him to specify it in the configuration.